### PR TITLE
Redesigning `protostar.toml` 12 — Rolling integration — `ProjectCompiler`

### PR DESCRIPTION
**Pull request by kasperski95**
Related #477

### CODE REVIEW

File: protostar/configuration_file/configuration_file_factory.py [hunk 1]
```diff
@@ -79,12 +79,12 @@ def _create_configuration_toml_v1(
 
     def _search_upwards_protostar_toml_path(self) -> Optional[Path]:
         directory_path = self._cwd
-        assert directory_path.exists()
         root_path = Path(directory_path.root)
         while directory_path != root_path:
-            for file_path in directory_path.iterdir():
-                if "protostar.toml" == file_path.name:
-                    return directory_path / "protostar.toml"
+            if directory_path.exists():
```
Comment (Arcticae) [inline protostar/configuration_file/configuration_file_factory.py:84]: I think we should throw if it doesn't exist

**Review by Arcticae — APPROVED** [review]

File: tests/integration/conftest.py [hunk 2]
```diff
@@ -208,10 +208,13 @@ def create_protostar_project():
             project_root_path=project_root_path,
             signing_credentials=signing_credentials,
         )
-
         protostar.init_sync()
         os.chdir(project_root_path)
-        yield protostar
+        yield build_protostar_fixture(
```
[the remark this answers is not in the document]
Comment (kasperski95) [reply tests/integration/conftest.py:213]: The configuration file is loaded during composing, and the configuration file changes after creating the project.

[the remark this answers is not in the document]
Comment (mkaput) [reply tests/integration/conftest.py:213]: could you leave a comment explaining this?

File: protostar/configuration_file/configuration_file_factory.py (hunk 1 above)
Comment (kasperski95) [reply protostar/configuration_file/configuration_file_factory.py:84]: Removed the check

File: tests/integration/conftest.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kasperski95) [reply tests/integration/conftest.py:217]: Refined `create_protostar_project`. Added a comment.

**Pull request merged**